### PR TITLE
Sebchw configurable training

### LINKED ISSUE #9: Create class for conducting experiments easily
Ideally such a class would take some configuration object containing information about:
- data
- task
- network
- hyperparameters
- logger
and then run experiment.

### PR DESCRIPTION
**Pull request by SebChw**
I checked few tools that allow easy configuration of ML experiments: [YACS](https://github.com/rbgirshick/yacs), [hydra](https://github.com/facebookresearch/hydra) and [spock](https://github.com/fidelity/spock). I chosen `hydra` since it has best support (it's maintained by facebook) and has ton of nice functionalities. Code I wrote was highly inspired by this [tutorial](https://www.kdnuggets.com/2023/03/hydra-configs-deep-learning-experiments.html) and [repo](https://github.com/gorodnitskiy/yet-another-lightning-hydra-template). I tried to take only most necessary thing from the repository so that it is easy for you to understand what is happening. In the future more stuff will be added

Additionaly I updated everything to torch and lightning 2.0 Since it was just few lines of code I do not open another PR.

closes #9

### CODE REVIEW

File: art/configs/logger/neptune.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+_target_: lightning.pytorch.loggers.NeptuneLogger
+api_key: ${oc.env:NEPTUNE_API_TOKEN} # api key is loaded from environment variable
```
Comment (kordc) [inline art/configs/logger/neptune.yaml:2]: Is the key uploaded in GH secrets, or everyone should use his API token? 
If second, should we write you our emails to give us access to the Neptune?

**Review by kordc — DISMISSED** [review]
LGTM

**Comment by SebChw** [comment]
Actually, there were two issues due to which mypy was failing:
* Firstly it seems that lightning 2.0 doesn't have stubs written yet. Also I needed to move mypy configurations to setup.cfg as toml file doesn't support syntax with asterisk.
* In one place I used import utils.logger instead of art.utils.logger and mypy tried to find `utils.logger` that doesn't exist and it was complaining

**Review by kordc — APPROVED** [review]

File: art/configs/logger/neptune.yaml (hunk 1 above)
Comment (SebChw) [reply art/configs/logger/neptune.yaml:2]: Everyone should use their own tokens. I'll let them now on the slack. Additionally I created Nepune organization `ghost-audio-team`. Yes you should write me emails connected with Neptune account.

**Pull request merged**